src/decode/lzma.rs

```diff
@@ -239,48 +197,53 @@ where
     Ok(decoder)
 }
 
-impl<W, LZB> DecoderState<W, LZB>
-where
-    W: io::Write,
-    LZB: lzbuffer::LzBuffer<W>,
-{
+impl DecoderState {
     pub fn reset_state(&mut self, lc: u32, lp: u32, pb: u32) {
+        if self.lc + self.lp == lc + lp {
```

Comment:
Likewise, the values of `lc`, `lp` and `pb` should be validated here.

Reply:
Done:

https://github.com/gendx/lzma-rs/pull/73/files#diff-768757151baad4fb0605aaeee5210dc343977125d64711ebcf5a0e755b2f23c0R177